sympy/assumptions/sathandlers.py

```diff
@@ -299,6 +315,19 @@ def register_fact(klass, fact, registry=fact_registry):
 for klass, fact in [
     (Mul, Equivalent(Q.zero, AnyArgs(Q.zero))),
     (MatMul, Implies(AllArgs(Q.square), Equivalent(Q.invertible, AllArgs(Q.invertible)))),
+    (MatAdd, Implies(AllArgs(Q.symmetric), Q.symmetric)),
+    (MatAdd, Implies(AllArgs(Q.diagonal), Q.diagonal)),
+    (MatMul, Implies(AllArgs(Q.diagonal), Q.diagonal)),
+    # TODO: or may be CustomLambda?
+    (MatrixExpr, CheckSquareMatrix(Q.square)),
+    (MatrixExpr, CheckOneByOneMatrix(Q.scalar_matrix)),
+    (MatrixExpr, CheckEmptyMatrix(Q.empty_matrix)),
+    (ZeroMatrix, Q.zero_matrix),
+    (Identity, Q.diagonal),
+    # TODO: why are tests failing if these rules are moved to ask.py?
+    (MatrixExpr, Implies(Q.scalar_matrix, Q.diagonal)),
```

Comment:
The problem was that I used `Equivalent` in the CheckOneByOneMatrix. Switching to `Implies` resolved the problem